File: Android/app/src/main/java/com/dreamproject/inwords/data/repository/translation/TranslationWordsLocalRepository.java
package com.dreamproject.inwords.data.repository.translation;

import com.dreamproject.inwords.data.entity.WordTranslation;

import java.util.List;

import io.reactivex.Completable;
import io.reactivex.Observable;
import io.reactivex.Single;

//Here any methods connected with manipulating data needed for Translation
public interface TranslationWordsLocalRepository {
    Observable<WordTranslation> getTranslation(String word);

    Observable<WordTranslation> getByOne();

    Observable<List<WordTranslation>> getList();

    Single<WordTranslation> addReplace(WordTranslation wordTranslation);

    Single<List<WordTranslation>> addReplaceAll(List<WordTranslation> wordTranslations);

    Completable removeAll(List<WordTranslation> wordTranslations);

    Completable removeAllServerIds(List<Integer> serverIds);
}

File: Android/app/src/main/java/ru/inwords/inwords/translation/data/repository/TranslationWordsLocalRepository.kt
package ru.inwords.inwords.translation.data.repository

import io.reactivex.Completable
import io.reactivex.Observable
import io.reactivex.Single
import ru.inwords.inwords.translation.data.bean.WordTranslation

//Here any methods connected with manipulating data needed for Translation
interface TranslationWordsLocalRepository {
    fun getList(): Observable<List<WordTranslation>>

    fun addReplace(wordTranslation: WordTranslation): Single<WordTranslation>

    fun addReplaceAll(wordTranslations: List<WordTranslation>): Single<List<WordTranslation>>

    fun removeAll(wordTranslations: List<WordTranslation>): Completable

    fun removeAllServerIds(serverIds: List<Int>): Completable

    fun clear()
}
